handler/src/main/java/io/netty/handler/ssl/util/FingerprintTrustManagerFactory.java

```diff
@@ -135,45 +131,97 @@ private void checkTrusted(String type, X509Certificate[] chain) throws Certifica
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints in hexadecimal form
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(Iterable<String> fingerprints) {
-        this(toFingerprintArray(fingerprints));
+        this("SHA1", toFingerprintArray(fingerprints));
     }
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints in hexadecimal form
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(String... fingerprints) {
-        this(toFingerprintArray(Arrays.asList(fingerprints)));
+        this("SHA1", toFingerprintArray(Arrays.asList(fingerprints)));
     }
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(byte[]... fingerprints) {
+        this("SHA1", fingerprints);
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param algorithm a hash algorithm
+     * @param fingerprints a list of fingerprints
+     */
+    private FingerprintTrustManagerFactory(final String algorithm, byte[][] fingerprints) {
+        ObjectUtil.checkNotNull(algorithm, "algorithm");
```

Comment:
should we call `MessageDigest.getInstance(algorithm)` once in the constructor now to ensure it will not fail later on ?